### PR TITLE
Eoehoma Firearms (& friends)

### PR DESCRIPTION
**Pull request by rye-rice**
## About The Pull Request

Screenshot_5451

Holy shit, this should not have taken a year to make

Adds the E-10, E-11, E-40, E-50, and E-60 to the game. Weapons manufactured by defunct firearms company Eoehoma Firearms.

Founded in 77 FS, Eoehoma was a early pioneer of ‘hybrid’ Solarian and Kalixcian laser weapons. The company went bankrupt due to increasingly poor and risky decision making, and all of it's patents were bought out by Nanotrasen. While Nanotrasen's Emitters bear a striking resemblance to the E-50, otherwise Nanotrasen has not produced any of Eoehoma's old weapons, instead focusing on Sharplite designed weapons.

Other changes:
- NT and Sharplite weapons have different fire sounds from each other
- Laser weapons buffed to 20 -> 25 damage
- Pulse shots don't destroy walls and are now 50 -> 40 damage
- Emitter shots now do 30 -> 60 damage
- Various grammar fixes
- Removes some non-lore compliant mentions
- Adds a manufacturer indicator to many guns
- Ports https://github.com/tgstation/tgstation/pull/60353
- Resprites various laser weaponry, notably the pulse guns.
- Deathsquad and ERT/LP hardsuits have been redone

## Why It's Good For The Game

image
image
image
image
ceredits to tetrazeta for the unfinished deathsquad sprite, i simply finished it and touched it up
image

One of the last things i remember the old leads planned was to buff lasers to make them stand up to the various ballistics better. Also allows Pulse Rifles to be more used in events by nerfing them to not be comedically overpowered. Now they are just Overpowered.

More ruin content and such. I'm sure the maptainers will make good use of this stuff.

And sprites, i fucking love sprites

## Changelog

:cl: retlaw34, tetrazeta
add: Eoehoma Firearms, a new guns manufacturer!
add: ERT and "Asset Protection" Hardsuits have gotten a new look!
add: New laser fire sounds

balance: Lasers now do slightly more damage
balance: Pulse rifles don't destroy walls anymore and do slightly less damage, and have lost their stun mode.
balance: Emitters do 60 damage and create turf fires on hitting a non-supermatter object.
fix: Various laser weapons that had broken autofire (E-TAR and the Tesla Cannon) now work

spellcheck: Grammar on some descriptions was corrected.
/:cl:

### CODE REVIEW

**Comment by Valorium-RB** [comment]
Woooooo! Congrats! Really glad to see all the hard work come to fruition.

File: code/modules/projectiles/guns/ballistic/assault.dm [hunk 1]
```diff
@@ -168,3 +169,109 @@
 	update_appearance()
 	for(var/datum/action/action as anything in actions)
 		action.UpdateButtonIcon()
+
+/obj/item/gun/ballistic/automatic/assualt/e40
+	name = "\improper E-40 Hybrid Rifle"
+	desc = "An Assualt Rifle, best known for being having a dual ballistic and laser system. Chambered in .299 eoehoma caseless, and uses energy for lasers."
```
Comment (Apogee-dev) [inline code/modules/projectiles/guns/ballistic/assault.dm]: "assualt" typo rears its head again, change all instances to "assault"
we should probably refactor and finally fix the same typo in this rifle class's path too

File: code/modules/projectiles/ammunition/ballistic/rifle.dm [hunk 2]
```diff
@@ -78,3 +78,10 @@
 	icon_state = "rifle-steel"
 	caliber = ".308 Winchester"
 	projectile_type = /obj/projectile/bullet/win308
+
+/obj/item/ammo_casing/caseless/c299
+	name = ".299 eoehoma caseless bullet casing"
```
Comment (Apogee-dev) [inline code/modules/projectiles/ammunition/ballistic/rifle.dm]: capitalize manufacturer name in cartridge e.g. ".229 Eoehoma caseless". find-and-replace to fix this across the codebase.

File: code/modules/projectiles/gun.dm [hunk 3]
```diff
@@ -2,6 +2,21 @@
...
 #define FIRING_PIN_REMOVAL_DELAY 50
 
+#define MANUFACTURER_NONE null
+#define MANUFACTURER_SHARPLITE "the Sharplite Defense logo"
+#define MANUFACTURER_SHARPLITE_NEW "the Nanotrasen-Sharplite logo"
+#define MANUFACTURER_HUNTERSPRIDE "the Hunter's Pride Arms and Ammunition logo"
+#define MANUFACTURER_SOLARARMORIES "the Solarbundswaffenkammer emblem"
+#define MANUFACTURER_SCARBOROUGH "the Scarborough Arms logo"
+#define MANUFACTURER_EOEHOMA "the Eoehoma Firearms emblem"
+#define MANUFACTURER_NANOTRASEN_OLD "the retro Nanotrasen logo"
```
Comment (Apogee-dev) [inline code/modules/projectiles/gun.dm]: You could probably describe this a little better, like "an outdated Nanotrasen logo"

File: code/modules/projectiles/gun.dm [hunk 4]
```diff
@@ -2,6 +2,21 @@
...
+#define MANUFACTURER_NANOTRASEN "the Nanotrasen logo"
+#define MANUFACTURER_URSA "the Major Ursa's Reserve logo"
+#define MANUFACTURER_BRAZIL "an green flag with a blue circle and a yellow diamond around it"
```
Comment (Apogee-dev) [inline code/modules/projectiles/gun.dm]: Grammar edit: "a green flag (...)"

File: code/modules/projectiles/gun.dm [hunk 5]
```diff
@@ -2,6 +2,21 @@
...
+#define MANUFACTURER_INTEQ "a orange crest with the letters 'IRMG'"
```
Comment (Apogee-dev) [inline code/modules/projectiles/gun.dm]: grammar edit: "an orange crest (...)"

File: code/modules/projectiles/guns/ballistic/revolver.dm [hunk 6]
```diff
@@ -74,7 +75,7 @@
 		. += "It can be spun with <b>alt+click</b>"
 
 /obj/item/gun/ballistic/revolver/detective
-	name = "\improper Colt Detective Special"
+	name = "\improper Hunters Pride Detective Special"
```
Comment (Apogee-dev) [inline code/modules/projectiles/guns/ballistic/revolver.dm]: Grammar edit, this and every other instance of that manufacturer name should be "Hunter's Pride"

File: code/modules/projectiles/guns/ballistic/shotgun.dm [hunk 7]
```diff
@@ -199,6 +205,7 @@
 	bolt_type = BOLT_TYPE_NO_BOLT
 	can_be_sawn_off  = TRUE
 	pb_knockback = 3 // it's a super shotgun!
+	manufacturer = MANUFACTURER_NONE
```
Comment (Apogee-dev) [inline code/modules/projectiles/guns/ballistic/shotgun.dm]: Shouldn't this be Hunter's Pride? It's right up their alley.

File: code/modules/projectiles/guns/energy/energy_gun.dm [hunk 8]
```diff
@@ -34,19 +35,31 @@
 	set_gun_light(new /obj/item/flashlight/seclite(src))
 	return ..()
 
-/obj/item/gun/energy/e_gun/stun
-	name = "tactical energy gun"
-	desc = "Military issue energy gun, is able to fire stun rounds."
+/obj/item/gun/energy/e_gun/hades
+	name = "SL AL-655 'Hades' energy rifle"
+	desc = "The standard issue rifle of the Nanotrasen Security Forces. Most have been mothballed into storage following the ICW, and aren't often issued to low ranking security divisions."
```
Comment (Apogee-dev) [inline code/modules/projectiles/guns/energy/energy_gun.dm]: Slight edit to iron out some awkward rhythm:
"The standard issue rifle of Nanotrasen's Security Forces. Most have been put in long term storage following the ICW, and usually aren't issued to low ranking security divisions."

File: code/modules/projectiles/guns/energy/energy_gun.dm [hunk 9]
```diff
@@ -34,19 +35,31 @@
...
+	slot_flags = ITEM_SLOT_BACK
+
+/obj/item/gun/energy/e_gun/hades/Initialize()
+	. = ..()
+	AddComponent(/datum/component/automatic_fire, 0.2 SECONDS)
 
 /obj/item/gun/energy/e_gun/old
 	name = "prototype energy gun"
-	desc = "NT-P:01 Prototype Energy Gun. Early stage development of a unique laser rifle that has multifaceted energy lens allowing the gun to alter the form of projectile it fires on command."
+	desc = "NT-P:01 Prototype Energy Gun. Early stage development of a unique laser rifle that has multifaceted energy lens allowing the gun to alter the form of projectile it fires on command. The project was a dud, and Nanotrasen later aquired Sharplite to suit it's laser weapon needs."
```
Comment (Apogee-dev) [inline code/modules/projectiles/guns/energy/energy_gun.dm]: Spelling and grammar edits. You get its/it's mixed up a lot, keep an eye out for it.

"NT-P:01 Prototype Energy Gun. Early stage development of a unique laser rifle that has a multifaceted energy lens, allowing the gun to alter the form of projectile it fires on command. The project was a dud, and Nanotrasen later acquired Sharplite to suit its laser weapon needs."

File: code/modules/projectiles/guns/energy/energy_gun.dm [hunk 10]
```diff
@@ -63,18 +76,21 @@
...
 /obj/item/gun/energy/e_gun/hos/brazil
 	name = "modified antique laser gun"
 	desc = "It's somehow modified to have more firemodes."
 	icon_state = "capgun_brazil_hos"
 	item_state = "hoslaser"
+	manufacturer = MANUFACTURER_SHARPLITE
 
 /obj/item/gun/energy/e_gun/hos/brazil/true
-	desc = "By using a original antique laser gun with a intact nuclear power cell, it is truly one of the finest weapons in this area of space."
+	desc = "By using a original antique laser gun with a intact nuclear power cell, it is truly one of the finest weapons in this sector."
```
Comment (Apogee-dev) [inline code/modules/projectiles/guns/energy/energy_gun.dm]: Rewrote this old description to actually be descriptive.

"This genuine antique laser gun, modified with an experimental suite of alternative firing modes based on the X-01 MultiPhase Energy Gun, is truly one of the finest weapons in this sector."

File: code/modules/projectiles/guns/energy/energy_gun.dm [hunk 11]
```diff
@@ -194,6 +211,10 @@
 	charge_sections = 3
 	weapon_weight = WEAPON_LIGHT
 
+/obj/item/gun/energy/e_gun/smg/Initialize()
+	. = ..()
+	AddComponent(/datum/component/automatic_fire, 0.2 SECONDS)
+
 /obj/item/gun/energy/e_gun/iot
 	name = "\improper E-SG 500 Second Edition"
 	desc = "A improved version of the E-SG 255. It now includes two firing modes, disable and kill, while still keeping that sweed integrated computer. Please note that the screen is right next to the switch mode button."
```
Comment (Apogee-dev) [inline code/modules/projectiles/guns/energy/energy_gun.dm]: I know you didn't touch this one but that typo was irritating me.

"A improved version of the E-SG 255. It now includes two firing modes, disable and kill, while still keeping that sweet integrated computer. Please note that the screen is right next to the switch mode button."

File: code/modules/projectiles/guns/energy/energy_gun.dm [hunk 12]
```diff
@@ -208,3 +229,17 @@
 	ammo_type = list(/obj/item/ammo_casing/energy/disabler/scatter/ultima, /obj/item/ammo_casing/energy/laser/ultima)
 	var/obj/item/modular_computer/integratedNTOS
 	var/NTOS_type = /obj/item/modular_computer/internal
+
+/obj/item/gun/energy/e_gun/e11
+	name = "E-11 hybrid energy rifle"
+	desc = "A hybrid energy gun fondly remembered as one of the worst weapons ever made. It hurts, but thats if you can even get hit with it."
```
Comment (Apogee-dev) [inline code/modules/projectiles/guns/energy/energy_gun.dm]: Slight grammar edit and some rewording for flow.

"A hybrid energy gun fondly remembered as one of the worst weapons ever made. It hurts, but that's only if it manages to hit its target."

File: code/modules/projectiles/guns/energy/laser.dm [hunk 13]
```diff
@@ -1,50 +1,46 @@
...
+	desc = "An antiquated model of the basic lasergun, no longer used or sold by Sharplite. Nevertheless, the sheer popularity of this model makes it a somewhat common sight to this day."
 	ammo_x_offset = 3
+	manufacturer = MANUFACTURER_SHARPLITE
 
 /obj/item/gun/energy/laser/captain
 	name = "antique laser gun"
 	icon_state = "caplaser"
 	item_state = null
-	desc = "This is an antique laser gun. All craftsmanship is of the highest quality. It is decorated with assistant leather and chrome. The object menaces with spikes of energy. On the item is an image of Space Station 13. The station is exploding."
+	desc = "This is the SL X-00, an antique laser gun that none know how to recreate. All craftsmanship is of the highest quality. It is decorated with assistant leather and chrome. The gun menaces with spikes of energy."
```
Comment (Apogee-dev) [inline code/modules/projectiles/guns/energy/laser.dm]: Slight wording update. Also readded the station image bit without actually naming SS13 for the sake of keeping the dwarf fortress gag intact. You can remove that if you want.

"This is the SL X-00, an antique laser gun, out of production for decades and well beyond anyone's capacity to recreate. All craftsmanship is of the highest quality. It is decorated with assistant leather and chrome. The gun menaces with spikes of energy. On the item is an image of a space station. The station is exploding."

File: code/modules/projectiles/guns/energy/laser.dm [hunk 14]
```diff
@@ -1,50 +1,46 @@
...
+	selfcharge = TRUE
 	resistance_flags = INDESTRUCTIBLE | LAVA_PROOF | FIRE_PROOF | ACID_PROOF
+	manufacturer = MANUFACTURER_SHARPLITE
 
 /obj/item/gun/energy/laser/captain/brazil
 	icon_state = "capgun_brazil"
 	item_state = "caplaser"
-	desc = "This is an antique laser gun. It seems all the high quality materials it was once made of are now scratched up and torn. The nuclear power cell has been removed, and will no longer automaticly recharge."
-	selfcharge = 0
+	desc = "This is the SL X-00, an antique laser gun that none know how to recreate. It seems all the high quality materials it was once made of are now scratched up and torn. The automatic recharging components has been removed, and will no longer automaticly recharge."
```
Comment (Apogee-dev) [inline code/modules/projectiles/guns/energy/laser.dm]: "This is the SL X-00, an antique laser gun, out of production for decades and well beyond anyone's capacity to recreate. It seems all the high quality materials it was once made of are now scratched up and torn. The nuclear power cell has been removed, and the gun will no longer automatically recharge."

File: code/modules/projectiles/guns/energy/laser.dm [hunk 15]
```diff
@@ -195,9 +198,38 @@
...
 /obj/item/gun/energy/laser/hitscanpistol/examine_more(mob/user)
 	if(in_range(src, user) || isobserver(user))
-		. = list("<span class='notice'>You examine [src] closer. Under the grip is a small inscription: \"NT CN SVALINN 2502\".</span>")
+		. = list("<span class='notice'>You examine [src] closer. Under the grip is a small inscription: \"NT CN SVALINN 462\".</span>")
 	else
 		. = list("<span class='warning'>You try to examine [src] closer, but you're too far away.</span>")
+
+/obj/item/gun/energy/laser/e10
+	name = "E-10 laser pistol"
+	desc = "A very old laser weapon. Despite the extreme age of some of these weapons, they are sometimes prefered to the mass produced Nanotrasen laser weapons."
```
Comment (Apogee-dev) [inline code/modules/projectiles/guns/energy/laser.dm]: Spelling edit.

"A very old laser weapon. Despite the extreme age of some of these weapons, they are sometimes preferred to newer, mass-produced Nanotrasen laser weapons."

File: code/modules/projectiles/guns/energy/laser.dm [hunk 16]
```diff
@@ -195,9 +198,38 @@
...
+	icon_state = "e10"
+	w_class = WEIGHT_CLASS_SMALL
+	ammo_type = list(/obj/item/ammo_casing/energy/lasergun/eoehoma)
+	manufacturer = MANUFACTURER_EOEHOMA
+
+/obj/item/gun/energy/laser/e50
+	name = "E-50 energy emitter"
+	desc = "A heavy and extremely powerful laser. Sets fires and kick ass, however uses a massive amount of energy per shot and is generally awkard to fire with."
```
Comment (Apogee-dev) [inline code/modules/projectiles/guns/energy/laser.dm]: Grammar, wording, and spelling edits. This _does_ start fires right?

"A heavy and extremely powerful laser. Sets targets on fire and kicks ass, but it uses a massive amount of energy per shot and is generally awkward to handle."

File: code/modules/projectiles/guns/energy/pulse.dm [hunk 17]
```diff
@@ -1,16 +1,20 @@
 /obj/item/gun/energy/pulse
 	name = "pulse rifle"
-	desc = "A heavy-duty, multifaceted energy rifle with three modes. Preferred by front-line combat personnel."
+	desc = "A heavy-duty, expensive, multifaceted energy rifle with three modes. The standard for heavy specialist Nanotrasen security."
```
Comment (Apogee-dev) [inline code/modules/projectiles/guns/energy/pulse.dm]: Wording and flow.

"A top-of-the-line, heavy-duty, multifaceted energy rifle with three firing modes. The gold standard for Nanotrasen's heavier security specialists."

File: code/modules/projectiles/guns/energy/pulse.dm [hunk 18]
```diff
@@ -33,41 +37,47 @@
...
 /obj/item/gun/energy/pulse/loyalpin
-	pin = /obj/item/firing_pin/implant/mindshield
+	pin = /obj/item/firing_pin
 
 /obj/item/gun/energy/pulse/carbine
 	name = "pulse carbine"
-	desc = "A compact variant of the pulse rifle with less firepower but easier storage."
-	w_class = WEIGHT_CLASS_NORMAL
-	slot_flags = ITEM_SLOT_BELT
+	desc = "The next-generation rifle of nanotrasen, however the logistical issues of making it the standard has left it only used by specialist LP and ERT teams."
```
Comment (Apogee-dev) [inline code/modules/projectiles/guns/energy/pulse.dm]: Reworded, grammar edits.

"A next-generation pulse weapon for Nanotrasen's security forces. High production costs and logistical issues have limited its deployment to specialist Loss Prevention and Emergency Response units."

**Review by Apogee-dev — DISMISSED** [review]
Mostly grammar, spelling, etc edits here. I'll leave balance to someone else for now.

File: code/modules/projectiles/guns/energy/laser.dm (hunk 13 above)
Comment (Valorium-RB) [reply code/modules/projectiles/guns/energy/laser.dm]: I would also like to argue that "assistant leather" be replaced with something less overtly bizarre, given Nanotrasen's marginally less deranged nature in our setting. Goliath leather, maybe.

Comment (rye-rice) [reply code/modules/projectiles/guns/energy/laser.dm]: Would work, but with how common goliath leather is im not sure if that would be considered 'antique.' Maybe a more rare mob? cant think of one though

Comment (Valorium-RB) [reply code/modules/projectiles/guns/energy/laser.dm]: Maybe a megafauna, like a drake?

Comment (rye-rice) [reply code/modules/projectiles/guns/energy/laser.dm]: yeah, that would work

File: code/modules/projectiles/guns/ballistic/shotgun.dm (hunk 7 above)
Comment (rye-rice) [reply code/modules/projectiles/guns/ballistic/shotgun.dm]: I always saw the double barreled shotgun as an indie weapon, like anyone could make one pretty easily, but yeah, I can make it HP

**Comment by rye-rice** [comment]
conflicts fixed

**Comment by rye-rice** [comment]
conlficts fixed... hopefully

**Comment by thgvr** [comment]
We should TM this, let me know next uptime if it's ready

**Review by thgvr — APPROVED** [review]
sprice

**Comment by rye-rice** [comment]
> We should TM this, let me know next uptime if it's ready

it should be ready, only review is from a maptainer

**Comment by rye-rice** [comment]
conflicts fixed, tm pleadster

**Comment by thgvr** [comment]
after gun pr i guess

File: code/modules/projectiles/guns/ballistic/assault.dm [hunk 19]
```diff
@@ -161,3 +162,109 @@
...
+		if(1)
+			select = 2
+			to_chat(user, "<span class='notice'>You switch to full auto laser.</span>")
+			SEND_SIGNAL(src, COMSIG_GUN_DISABLE_AUTOFIRE)
+			SEND_SIGNAL(secondary, COMSIG_GUN_ENABLE_AUTOFIRE)
+		if(2)
+			select = 1
+			to_chat(user, "<span class='notice'>You switch to full automatic ballistic.</span>")
+			SEND_SIGNAL(src, COMSIG_GUN_ENABLE_AUTOFIRE)
+			SEND_SIGNAL(secondary, COMSIG_GUN_DISABLE_AUTOFIRE)
```
Comment (MarkSuckerberg) [inline code/modules/projectiles/guns/ballistic/assault.dm]: the values of `select` should probably be made into defines

File: code/modules/projectiles/guns/ballistic/assault.dm [hunk 20]
```diff
@@ -161,3 +162,109 @@
...
+		return ..()
+
+/obj/item/gun/ballistic/automatic/assualt/e40/process_fire(atom/target, mob/living/user, message, params, zone_override, bonus_spread)
+	. = ..()
+	if(select == 2)
+		secondary.process_fire(target, user, message, params, zone_override, bonus_spread)
+	else
+		return ..()
+
+
```
Comment (MarkSuckerberg) [inline code/modules/projectiles/guns/ballistic/assault.dm]:
```suggestion

/obj/item/gun/ballistic/automatic/assualt/e40/can_fire()
	if(select == 2)
		return secondary.can_fire()
	else
		return ..()

```

File: code/modules/projectiles/guns/ballistic/assault.dm [hunk 21]
```diff
@@ -161,3 +162,109 @@
...
+	update_appearance()
+
+/obj/item/gun/ballistic/automatic/assualt/e40/do_autofire(datum/source, atom/target, mob/living/shooter, params)
+	if(select == 2)
+		secondary.do_autofire(source, target, shooter, params)
+	else
+		return ..()
+
+/obj/item/gun/ballistic/automatic/assualt/e40/process_fire(atom/target, mob/living/user, message, params, zone_override, bonus_spread)
+	. = ..()
```
Comment (MarkSuckerberg) [inline code/modules/projectiles/guns/ballistic/assault.dm]:
```suggestion
```

File: code/modules/projectiles/guns/ballistic/assault.dm [hunk 22]
```diff
@@ -161,3 +162,109 @@
...
+	name = "secondary e40 laser gun"
+	desc = "The laser component of a E-40 Hybrid Rifle. You probably shouldn't see this."
+	fire_sound = 'sound/weapons/gun/laser/e40_las.ogg'
+	w_class = WEIGHT_CLASS_NORMAL
+	ammo_type = list(/obj/item/ammo_casing/energy/laser/assault)
+	fire_delay = 2
+
+/obj/item/gun/energy/laser/e40_laser_secondary/Initialize()
+	. = ..()
+	AddComponent(/datum/component/automatic_fire, 0.2 SECONDS)
```
Comment (MarkSuckerberg) [inline code/modules/projectiles/guns/ballistic/assault.dm]:
```suggestion
```

File: code/modules/projectiles/guns/ballistic/assault.dm [hunk 23]
```diff
@@ -161,3 +162,109 @@
...
+	else
+		return ..()
+
+/obj/item/gun/ballistic/automatic/assualt/e40/attackby(obj/item/attack_obj, mob/user, params)
+	if(istype(attack_obj, /obj/item/stock_parts/cell/gun) ||(istype(attack_obj, /obj/item/screwdriver)))
+		secondary.attack_self()
+		secondary.attackby(attack_obj, user, params)
+	else
+		..()
+
```
Comment (MarkSuckerberg) [inline code/modules/projectiles/guns/ballistic/assault.dm:263]:
```suggestion

/obj/item/gun/ballistic/automatic/assualt/e40/can_shoot()
	if(select == 2)
		return secondary.can_shoot()
	return ..()

```

**Review by MarkSuckerberg — DISMISSED** [review]

File: code/modules/projectiles/guns/ballistic/assault.dm [hunk 24]
```diff
@@ -161,3 +162,114 @@
...
+	if(select == 2)
+		secondary.process_fire(target, user, message, params, zone_override, bonus_spread)
+	else
+		return ..()
+
+/obj/item/gun/ballistic/automatic/assualt/e40/can_fire()
+	if(select == 2)
+		return secondary.can_fire()
+	else
+		return ..()
```
Comment (MarkSuckerberg) [inline code/modules/projectiles/guns/ballistic/assault.dm]:
```suggestion
/obj/item/gun/ballistic/automatic/assualt/e40/can_shoot()
	if(select == 2)
		return secondary.can_shoot()
	else
		return ..()
```
oopsie

**Comment by rye-rice** [comment]
conflict deez nuts

File: code/modules/projectiles/guns/ballistic/assault.dm [hunk 25]
```diff
@@ -191,3 +192,123 @@
...
+	mag_display = TRUE
+	empty_indicator = TRUE
+	fire_sound = 'sound/weapons/gun/laser/e40_bal.ogg'
+	manufacturer = MANUFACTURER_EOEHOMA
+
+/obj/item/gun/ballistic/automatic/assault/e40/Initialize()
+	. = ..()
+	secondary = new /obj/item/gun/energy/laser/e40_laser_secondary(src)
+	AddComponent(/datum/component/automatic_fire, 0.2 SECONDS)
+	RegisterSignal(secondary, COMSIG_ATOM_UPDATE_ICON, .proc/secondary_update_icon)
```
Comment (thgvr) [inline code/modules/projectiles/guns/ballistic/assault.dm]:
```suggestion
	RegisterSignal(secondary, COMSIG_ATOM_UPDATE_ICON, PROC_REF(secondary_update_icon))
```

**Review by thgvr — CHANGES_REQUESTED** [review]
oops

**Review by thgvr — APPROVED** [review]

**Pull request merged**